comments.js

```diff
@@ -0,0 +1,42 @@
+import { ADD_COMMENT, EDIT_COMMENT, REMOVE_COMMENT, THUMB_UP_COMMENT, THUMB_DOWN_COMMENT } from './actions';
+
+const initialState = {
+    comments: [],
+    users: []
+};
+
+
+export const comments = (state = initialState, action) => {
+
+    switch (action.type) {
+
+        case ADD_COMMENT:
+            return [{
+                id: action.id,
+                text: action.text,
+                votesUp: 0,
+                votesDown: 0
+            }, ...state.comments];
+            break;
+
+        case EDIT_COMMENT:
+            return state.comments.find(comment => (comment.id === action.id)).text = action.text;
+            break;
+
+        case THUMB_UP_COMMENT:
+            return state.comments.find(comment => (comment.id === action.id)).votesUp++;
```

Comment:
w przypadku kiedy nie zostanie znaleziony komentarz z danym ID kod zwróci błąd.
Dobrze by było najpierw sprawdzić czy dany komentarz istnieje a dopiero potem odwoływać się do jego właściwości (votesUp)

Co więcej w Reduxie stosuje się taką metodę aby nie zmieniać istniejącego stanu / obiektu tylko zwracać nowy. W tym przypadku będzie to trudne ponieważ używasz listy do przechowywania komentarzy. Musiałbyś więc najpierw pobrać indeks elementu który spełnia kryteria wyszukiwania i później zrobić coś takiego:
```
return [
   ...comments.slice(0, index),
   {..comments[index], {votesUp: comments[index].votesUp + 1},
   ...comments.slice(index+1, comments.length)
];

Wygląda to dość szalenie ale zaleta jest taka że nie modyfikujesz obiektów które istnieją tylko zwracasz nowe. Dzięki temu react skuteczniej będzie w stanie zdecydować kiedy powinien zrobić "rerender".

Gdybyś natomiast zamienił listę komentarzy na obiekt o takiej strukturze:
```
{
   id: {text: ..., id: ...},
   ...
}
```

to mógłbyś zrobić coś takiego:
```
return {
   ...comments,
   [action.id]: {
      ...comments[action.id]
      votesUp: comments[action.id]++
   }
}
```